priv/zipgateway.tun

```diff
@@ -15,12 +15,19 @@ case "$1" in
       rm -f $ROUTING_PID
     fi
 
-    sh -c "while true; do
-               $PIDOF zipgateway || break;
+    sh -c "for i in 1 2 3 4 5 6; do
+               if [ \$i -gt 5 ]; then
+                 echo \"Failed to start tunnel script\"
+                 kill -9 \$PPID
```

Comment:
This $PPID is the pid of the `zipgateway.tun` shell script, so it has already exited. It would have to be gotten at the top. For example,

```
ZIPGATEWAY_PID=$PID
...
      echo \"Failed to start tunnel script \"
      kill $ZIPGATEWAY_PID
      exit 1
```
Note the exit 1. The exit status is unsigned. Exit -1 would work, but it looks weird.